versions/development/SPEC.md

```diff
@@ -1218,81 +1270,316 @@ $runtime = 'runtime' $ws* '{' ($ws* $runtime_kv $ws*)* '}'
+    cpu: 8
   }
 }
+
 ```
 
-#### memory
+#### memory (**Required**)
 
-Memory requirements for this task.  Two kinds of values are supported for this attributes:
+The `memory` key defines the _minimum_ memory required for this task which must be available prior to the engine starting execution. The engine does not need to provide the exact amount of memory requested, however it may ONLY provision more memory then requested and not less. For example, if the wdl requested `1 GB` but only blocks of `4 GB` were available, the engine might choose to provision `4.0 GB` instead.  Two kinds of values are supported for this attribute:
 
 * `Int` - Interpreted as bytes
 * `String` - This should be a decimal value with suffixes like `B`, `KB`, `MB` or binary suffixes `KiB`, `MiB`.  For example: `6.2 GB`, `5MB`, `2GiB`.
 
 ```wdl
 task memory_test {
-  input {
-    String arg
+  #....
+  runtime {
+    memory: "2 GB"
   }
+}
+```
 
-  command {
-    python process.py ${arg}
+#### gpu (**Optional**)
+
+The `gpu` key provides a way to accommodate modern workflows which are increasingly becoming reliant on GPU computations. This key is a `true` or `false` value and simply indicates to the engine that a task requires a GPU to run to completion. A task with this flag set to `true` is guaranteed to only run if a GPU is a available within the runtime environment. It is the responsibility of the engine to check prior to execution whether a GPU is provisionable, and if not, preemptively fail the task.
+
+It is important to note, that this flag does not provide any information on the quantity or type of GPUs to make available to the task. This information should be provided within the `hints` section according to the specific engine parameters.
+
+
+```wdl
+task gpu_test {
+  #.....
+  runtime {
+    gpu: true
   }
+}
+```
+
+
+#### disks (**Optional - but strongly encouraged**)
+
+The `disks` key provides a way to request one or more persistent volumes of at least a specific size and mounted at a specific location. When the disks key is provided the engine must guarantee the resources requested are available, or immediately fail the task prior to execution. This property does not specify exactly what type of persistent volume is being requested (ie SSD, HDD), but leaves this up to the engine to decide, based on what hardware is available or a `hints` value.
+
+The general format of a `disks` value is: `</absolute/mount/point> <Size in GB>`. A WDL may leave out the mount point and only provide the size in GB that is being requested for ONE disks entry. This should be interpreted by the engine as a persistent volume mounted at the root of the execution directory within a task. If more then one disks entries for a task leaves out the mount point, the engine should fail the tasks.
+
+There are several ways to specify the `disks` attribute:
+
+* `Int` - GB of disk space to request, eg: `100`, `200`.
+* `String` (`"<size>" || "<mount-point> <size>"`)
```

Comment:
Should some system similar to the one for memory be allowed here? ie. specifying whether the size is in KBs or MBs or TBs.

Reply:
Semantically that makes sense. A disk is a a size, and sizes can be specified in blocks. In practice however I feel like we will only ever see `GB` or `TB` definitions (since specifying a disk in MB seems archaic). And from that we will NORMALLY only see `GB`. So I figured the simplification makes sense. Wheras with memory, people can and regularily do specify both GB and MB for the size

Reply:
To clarify, are disk specifications like `"100GB`, or `"3TB"` valid? 

In my personal opinion, they should be legal.